### PR TITLE
1774/preferred unit

### PR DESCRIPTION
**Pull request by seanmalbert**
# Pull Request Template

## Issue

Addresses #1774 

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This only addresses the backend changes necessary for converting preferredUnit from string[] to UnitType[]

### CODE REVIEW

**Comment by seanmalbert** [comment]
@dominikx96, this adds the backend changes for 1774. Can you use this to update the frontend apps?

File: backend/core/src/applications/entities/application.entity.ts [hunk 1]
```diff
@@ -191,12 +194,10 @@ export class Application extends AbstractEntity {
...
-  @Column({ type: "text", array: true })
-  @Expose()
-  @IsString({ groups: [ValidationsGroupsEnum.default], each: true })
-  @ArrayMaxSize(8, { groups: [ValidationsGroupsEnum.default] })
-  @MaxLength(64, { groups: [ValidationsGroupsEnum.default], each: true })
-  preferredUnit: string[]
+  @ManyToMany(() => UnitType, { eager: true, cascade: true })
+  @JoinTable()
+  @Type(() => UnitType)
+  preferredUnit: UnitType[]
```
Comment (pbn4) [inline backend/core/src/applications/entities/application.entity.ts:200]: Does it work without an explicit `@Expose()`?

**Review by pbn4 — APPROVED** [review]

**Comment by seanmalbert** [comment]
@dominikx96 , this should be good to bring into your branch.

**Comment by seanmalbert** [comment]
Closing since I added this to a branch on upstream with #1778.

**Pull request closed without merging**